SMLHelper/Patchers/CraftDataPatcher.cs

```diff
@@ -34,6 +34,7 @@ public static void Patch(HarmonyInstance harmony)
             Utility.PatchDictionary(CraftDataType, "harvestTypeList", customHarvestTypeList);
             Utility.PatchDictionary(CraftDataType, "itemSizes", customItemSizes);
             Utility.PatchDictionary(CraftDataType, "equipmentTypes", customEquipmentTypes);
+            Utility.PatchDictionary(CraftDataType, "groups", customGroups);
```

Comment:
Now this can be used like that (example from your BaseLightSwitch)
```
            if(!CraftDataPatcher.customGroups.ContainsKey(TechGroup.InteriorModules))
                CraftDataPatcher.customGroups.Add(TechGroup.InteriorModules, new Dictionary<TechCategory, List<TechType>>());
            if(!CraftDataPatcher.customGroups[TechGroup.InteriorModules].ContainsKey(TechCategory.InteriorModule))
                CraftDataPatcher.customGroups[TechGroup.InteriorModules][TechCategory.InteriorModule] = new List<TechType>();
            CraftDataPatcher.customGroups[TechGroup.InteriorModules][TechCategory.InteriorModule].Add(techType);```